### PR TITLE
Migrate to machine-api-provider-aws

### PR DESCRIPTION
**Pull request by alexander-demicev**
In order to migrate from cluster-api-provider-aws to machine-api-provider-aws I added similar test suite to new repo and disabled image promotion in cluster-api-provider-aws.

### CODE REVIEW

File: ci-operator/config/openshift/machine-api-provider-aws/openshift-machine-api-provider-aws-main.yaml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  commands: IS_CONTAINER=TRUE make goimports
+  container:
+    from: src
+  skip_if_only_changed: ^docs/|\.md$|^(?:.*/)?(?:\.gitignore|OWNERS|PROJECT|LICENSE)$
+- as: govet
+  commands: IS_CONTAINER=TRUE make vet
+  container:
+    from: src
+  skip_if_only_changed: ^docs/|\.md$|^(?:.*/)?(?:\.gitignore|OWNERS|PROJECT|LICENSE)$
+- as: generate
```
Comment (lobziik) [inline ci-operator/config/openshift/machine-api-provider-aws/openshift-machine-api-provider-aws-main.yaml:56]: Should we delete this generate task probably? this was moved to openshift/api, isn't it?

**Comment by lobziik** [comment]
/lgtm
/hold
Would left a space for another review.

**Comment by alexander-demichev** [comment]
@JoelSpeed can you lgtm again? I had to run `make jobs` one more time.

**Pull request merged**